File: app/src/main/java/com/xattacker/android/bingo/CustomProperties.java
package com.xattacker.android.bingo;

import android.content.Context;
import android.view.Display;

import com.xattacker.android.bingo.util.AppProperties;

import java.util.Hashtable;

public final class CustomProperties
{
	private static Hashtable<FontType, Integer> FONT_SIZE = new Hashtable<FontType, Integer>();

	public static int getDimensionPxSize(FontType aType, Context aContext)
	{
		int size = 0;
		
		if (FONT_SIZE.contains(aType))
		{
			size = FONT_SIZE.get(aType);
		}
		else // not found, create new one
		{
			int res = 0;
			
			switch (aType)
			{
				case VERY_LARGE_FONT_SIZE:
					res = R.dimen.VERY_LARGE_FONT_SIZE;
					break;
					
				case LARGE_FONT_SIZE:
					res = R.dimen.LARGE_FONT_SIZE;
					break;
					
				case LARGE_NORMAL_FONT_SIZE:
					res = R.dimen.LARGE_NORMAL_FONT_SIZE;
					break;
					
				case NORMAL_FONT_SIZE:
					res = R.dimen.NORMAL_FONT_SIZE;
					break;
					
				case SMALL_NORMAL_FONT_SIZE:
					res = R.dimen.SMALL_NORMAL_FONT_SIZE;
					break;
					
				case SMALL_FONT_SIZE:
					res = R.dimen.SMALL_FONT_SIZE;
					break;
					
				case VERY_SMALL_FONT_SIZE:
					res = R.dimen.VERY_SMALL_FONT_SIZE;
					break;
			}

			size = aContext.getResources().getDimensionPixelSize(res);
			FONT_SIZE.put(aType, size);
		}
		
		return size;
	}
	
	public static int getScreenWidth(float aRatio)
	{
		int width = 0;
		
		Display display = AppProperties.getScreenDisplay();
		if (display != null)
		{
			width = (int)(display.getWidth() * aRatio);
		}
		
		return width;
	}
	
	public static int getScreenHeight(float aRatio)
	{
		int height = 0;
		
		Display display = AppProperties.getScreenDisplay();
		if (display != null)
		{
			height = (int)(display.getHeight() * aRatio);
		}
		
		return height;
	}
}

File: app/src/main/java/com/xattacker/android/bingo/CustomProperties.kt
package com.xattacker.android.bingo

import android.content.Context
import android.util.DisplayMetrics
import android.view.Display
import com.xattacker.android.bingo.util.AppProperties
import java.util.Hashtable

object CustomProperties
{
    private val FONT_SIZE = Hashtable<FontType, Int>()

    fun getDimensionPxSize(aType: FontType, aContext: Context): Int
    {
        var size = 0

        if (FONT_SIZE.contains(aType))
        {
            size = FONT_SIZE[aType] ?: 0
        }
        else // not found, create new one
        {
            var res = 0

            when (aType)
            {
                FontType.VERY_LARGE_FONT_SIZE -> res = R.dimen.VERY_LARGE_FONT_SIZE
                FontType.LARGE_FONT_SIZE -> res = R.dimen.LARGE_FONT_SIZE
                FontType.LARGE_NORMAL_FONT_SIZE -> res = R.dimen.LARGE_NORMAL_FONT_SIZE
                FontType.NORMAL_FONT_SIZE -> res = R.dimen.NORMAL_FONT_SIZE
                FontType.SMALL_NORMAL_FONT_SIZE -> res = R.dimen.SMALL_NORMAL_FONT_SIZE
                FontType.SMALL_FONT_SIZE -> res = R.dimen.SMALL_FONT_SIZE
                FontType.VERY_SMALL_FONT_SIZE -> res = R.dimen.VERY_SMALL_FONT_SIZE
            }

            size = aContext.resources.getDimensionPixelSize(res)
            FONT_SIZE[aType] = size
        }

        return size
    }

    fun getScreenWidth(aRatio: Float): Int
    {
        var width = 0
        val display = AppProperties.screenDisplay
        if (display != null)
        {
            val dm = DisplayMetrics()
            display.getMetrics(dm)
            width = (dm.widthPixels.toFloat() * aRatio).toInt()
        }

        return width
    }

    fun getScreenHeight(aRatio: Float): Int
    {
        var height = 0
        val display = AppProperties.screenDisplay
        if (display != null)
        {
            val dm = DisplayMetrics()
            display.getMetrics(dm)
            height = (dm.heightPixels.toFloat() * aRatio).toInt()
        }

        return height
    }
}
